Mathlib/MeasureTheory/Group/Integral.lean

```diff
@@ -112,7 +112,8 @@ to a left-invariant measure is 0. -/
       respect to a left-invariant measure is 0. -/]
 theorem integral_eq_zero_of_mul_left_eq_neg [IsMulLeftInvariant μ] (hf' : ∀ x, f (g * x) = -f x) :
     ∫ x, f x ∂μ = 0 := by
-  simp_rw [← self_eq_neg ℝ E, ← integral_neg, ← hf', integral_mul_left_eq_self]
+  have := noZeroSMulDivisors_nat_iff_isAddTorsionFree.1 <| Nat.noZeroSMulDivisors ℝ E
```

Comment:
Note: this is ugly, but can't easily be made nicer without essentially redoing #30563

Reply:
I don't see the point here. The cleanup will be done in #30563, and there is no way to miss it (because I will be deprecating `noZeroSMulDivisors_nat_iff_isAddTorsionFree` there).